src/Symfony/Component/DependencyInjection/Compiler/RepeatedPass.php

```diff
@@ -56,14 +56,12 @@ public function __construct(array $passes)
      */
     public function process(ContainerBuilder $container)
     {
-        $this->repeat = false;
-        foreach ($this->passes as $pass) {
-            $pass->process($container);
-        }
-
-        if ($this->repeat) {
-            $this->process($container);
-        }
+        do {
+            $this->repeat = false;
+            foreach ($this->passes as $pass) {
+                $pass->process($container);
+            }
+        } while ($this->repeat);
```

Comment:
if a `RepeatedPass` unvariably sets `$this->repeat` to `true`, then this will turn into an infinite loop, isn't it? Or are we safe from this issue?